File: OsmAndWear/src/net/osmand/plus/track/GpxSplitType.java
package net.osmand.plus.track;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;

import net.osmand.plus.R;

public enum GpxSplitType {

	NO_SPLIT("no_split", -1, R.string.shared_string_none),
	DISTANCE("distance", 1, R.string.distance),
	TIME("time", 2, R.string.shared_string_time);

	private final String typeName;
	private final int type;
	@StringRes
	private final int resId;

	GpxSplitType(@NonNull String typeName, int type, @StringRes int resId) {
		this.typeName = typeName;
		this.type = type;
		this.resId = resId;
	}

	public int getType() {
		return type;
	}

	public String getTypeName() {
		return typeName;
	}

	public String getHumanString(@NonNull Context ctx) {
		return ctx.getString(resId);
	}

	@NonNull
	public static GpxSplitType getSplitTypeByName(@Nullable String name) {
		for (GpxSplitType splitType : values()) {
			if (splitType.name().equalsIgnoreCase(name)) {
				return splitType;
			}
		}
		return NO_SPLIT;
	}

	@NonNull
	public static GpxSplitType getSplitTypeByTypeId(int typeId) {
		for (GpxSplitType splitType : values()) {
			if (splitType.type == typeId) {
				return splitType;
			}
		}
		return NO_SPLIT;
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/gpx/GpxSplitType.kt
package net.osmand.shared.gpx

import net.osmand.shared.util.Localization

enum class GpxSplitType(val typeName: String, val type: Int, val resId: String) {
	NO_SPLIT("no_split", -1, "shared_string_none"),
	DISTANCE("distance", 1, "distance"),
	TIME("time", 2, "shared_string_time");

	companion object {
		fun getSplitTypeByName(name: String?): GpxSplitType {
			return entries.find { it.name.equals(name, ignoreCase = true) } ?: NO_SPLIT
		}

		fun getSplitTypeByTypeId(typeId: Int): GpxSplitType {
			return entries.find { it.type == typeId } ?: NO_SPLIT
		}
	}

	fun getHumanString(): String {
		return Localization.getString(resId)
	}
}

